agent/src/agent/destination/repository.py

```diff
@@ -0,0 +1,49 @@
+from agent.db import Session, entity
+from agent.destination import HttpDestination
+
+session = Session()
+
+
+def exists() -> bool:
+    return session.query(
+        session.query(entity.Destination).exists()
+    ).scalar()
+
+
+def get() -> HttpDestination:
+    return HttpDestination.from_entity(_get_entity())
+
+
+def upsert(destination_: HttpDestination):
+    update(destination_) if exists() else create(destination_)
+
+
+def create(destination_: HttpDestination):
+    session.add(destination_.to_entity())
+    session.commit()
+
+
+def update(destination_: HttpDestination):
```

Comment:
For example, if you add a new column to the entity you will have to not forget to set it in two places, here and in the `to_entity` method. The same is for sources and pipelines. Maybe it's better to store the entity object inside the main object? Then `to_entity()` function will return a new entity if not exists and updated entity otherwise